syntax = "proto3";

service Movie {
    rpc GetMovieByID(MovieID) returns (MovieData) {}
    rpc GetListMovies(Empty) returns (stream MovieData) {}
<<<<<<< HEAD
    rpc GetMovieByTitle(MovieTitle) returns (MovieData) {}
    rpc GetMovieRate(MovieID) returns (MovieRate) {}
    rpc AddMovie(MovieData) returns (OperationSuccess){}
    rpc ModifyMovie(MovieData) returns (OperationSuccess){}
    rpc DeleteMovie(MovieID) returns (OperationSuccess){}
=======
>>>>>>> 6b88be6259f92c034ff7cfd82b22c2034cb337af
}

message MovieID {
    string id = 1;
}

message MovieTitle{
    string title = 1;
}

message MovieRate{
    float rating = 1;
}

message MovieData {
    string title = 1;
    float rating = 2;
    string director = 3;
    string id = 4;
}

<<<<<<< HEAD
message OperationSuccess{
    bool success = 1;
    string comment = 2;
}

 message Empty{

 }
=======
message Empty {

}
>>>>>>> 6b88be6259f92c034ff7cfd82b22c2034cb337af
